test/services/company.profile.service.unit.ts

```diff
@@ -0,0 +1,138 @@
+  }
+});
+
+describe("Company profile service test", () => {
+  const COMPANY_NUMBER = "1234567";
+
+  beforeEach(() => {
+    mockGetCompanyProfile.mockReset();
+    mockLoggerError.mockClear();
+    mockCreateApiClient.mockClear();
+  });
+
+  it("Should return a company profile", async () => {
+    mockGetCompanyProfile.mockResolvedValueOnce(dummySDKResponse);
+    const returnedProfile = await getCompanyProfile(COMPANY_NUMBER);
+
+    expect(returnedProfile).toBe(dummySDKResponse.resource);
+  });
+
+  it("Should throw an error if status code == 400", () => {
+    const HTTP_STATUS_CODE = 400;
+    mockGetCompanyProfile.mockResolvedValueOnce({
+      httpStatusCode: HTTP_STATUS_CODE
+    } as Resource<CompanyProfile>);
+
+    getCompanyProfile(COMPANY_NUMBER)
+      .then(() => {
+        fail("Was expecting an error to be thrown.");
+      })
+      .catch((error: Error) => {
+        expect(error.message).toContain(HTTP_STATUS_CODE);
+        expect(error.message).toContain(COMPANY_NUMBER);
+        expect(mockLoggerError).toHaveBeenCalled();
+      });
+  });
+
+  it("Should throw an error if status code > 400", () => {
+    const HTTP_STATUS_CODE = 502;
+    mockGetCompanyProfile.mockResolvedValueOnce({
+      httpStatusCode: HTTP_STATUS_CODE
+    } as Resource<CompanyProfile>);
+
+    getCompanyProfile(COMPANY_NUMBER)
+      .then(() => {
+        fail("Was expecting an error to be thrown.");
+      })
+      .catch((error: Error) => {
+        expect(error.message).toContain(HTTP_STATUS_CODE);
+        expect(error.message).toContain(COMPANY_NUMBER);
+        expect(mockLoggerError).toHaveBeenCalled();
+      });
+  });
+
+  it("Should throw an error if no response returned from SDK", () => {
+    mockGetCompanyProfile.mockResolvedValueOnce(undefined);
+
+    getCompanyProfile(COMPANY_NUMBER)
+      .then(() => {
+        fail("Was expecting an error to be thrown.");
+      })
+      .catch((error: Error) => {
+        expect(error.message).toContain("no response");
+        expect(error.message).toContain(COMPANY_NUMBER);
+        expect(mockLoggerError).toHaveBeenCalled();
+      });
+  });
+
+  it("Should throw an error if no response resource returned from SDK", () => {
+    mockGetCompanyProfile.mockResolvedValueOnce({} as Resource<CompanyProfile>);
+
+    getCompanyProfile(COMPANY_NUMBER)
+      .then(() => {
+        fail("Was expecting an error to be thrown.");
+      })
+      .catch((error: Error) => {
+        expect(error.message).toContain("no resource");
+        expect(error.message).toContain(COMPANY_NUMBER);
+        expect(mockLoggerError).toHaveBeenCalled();
+      });
+  });
+});
+
+const dummySDKResponse: Resource<CompanyProfile> = {
+  httpStatusCode: 200,
+  resource: {
+    accounts: {
+      nextAccounts: {
+        periodEndOn: "2019-10-10",
+        periodStartOn: "2019-01-01",
+      },
+      nextDue: "2020-05-31",
+      overdue: false,
+    },
+    companyName: "Girl's school trust",
```

Comment:
should we change this data to not be an old copy of live data. 
Maybe change the company number to 12345678, and the company name to "Test Company"

Reply:
new version pushed